### PR TITLE
DB Connection using .env

### PR DESCRIPTION
**Pull request by Mohamedegypt**

### CODE REVIEW

File: backend/server/index.js [hunk 1]
```diff
@@ -1,13 +1,33 @@
...
+
+// Connection to DB using .env File
+mongoose.connect(process.env.DB_URI);
+var db = mongoose.connection;
+ console.log(process.env.DB_URI);
+db.once("open", () => {
+	console.log("DB Connection esthablised");
+});
 
+db.on("error", (errr) => {
```
Comment (PatrickS83) [inline backend/server/index.js]: - errr has one to many 'r'

File: backend/server/index.js [hunk 2]
```diff
@@ -1,13 +1,33 @@
 const express = require('express');
 const app = express();
-const PORT = process.env.PORT || 4000;
 const mockupJson = require("./mockup/mockup.json");
+const mongoose = require('mongoose');
```
Comment (PatrickS83) [inline backend/server/index.js]: 'mongoose' should be listed in the project's dependencies.

File: backend/server/index.js [hunk 3]
```diff
@@ -1,13 +1,33 @@
...
+const bodyParser = require('body-parser');
+const dotenv = require('dotenv').config();
+
+// Using Body Parser
+app.use(bodyParser());
+
+// Connection to DB using .env File
+mongoose.connect(process.env.DB_URI);
+var db = mongoose.connection;
```
Comment (PatrickS83) [inline backend/server/index.js]: please use const instead of var

**Comment by NachoMerino** [comment]
Reviewing, will take time

**Comment by NachoMerino** [comment]
You need to install and save the dependecies mongodb and mongoose

npm install --save mongoose mongodb

File: backend/server/index.js [hunk 4]
```diff
@@ -1,13 +1,106 @@
...
+  	console.log(req.body);
+  	res.send("Logged in!");
+  	res.redirect("/");
+ });
+
+// Lougout
+app.get('/logout', users.logout);
+
+// Server Port
+const PORT = process.env.PORT || 4000;
```
Comment (oliverwebr) [inline backend/server/index.js]: should be defined at the top please see issue #32

**Pull request closed without merging**